### PR TITLE
feat: throw invalid user input error

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/dotnet-deps-parser/blob/master/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Catch and re-throw known InvalidUserInput error for better errors in upstreams

### CODE REVIEW

**Comment by dkontorovskyy** [comment]
Not sure should it be called `InvalidUserInputError` if `JSON.parse` fails. Should be more descriptive that JSON is not a valid JSON or something similar. Thoughts?

**Comment by lili2311** [comment]
Testing locally to validate if any upstream changes are needed

**Comment by lili2311** [comment]
Tests look good @dkontorovskyy please take another look

File: lib/index.ts [hunk 1]
```diff
@@ -148,40 +165,51 @@ async function extractTargetFrameworksFromProjectConfig(
...
   try {
     // trimming required to address files with UTF-8 with BOM encoding
-    const manifestFile = JSON.parse(manifestFileContents.trim());
-    return getTargetFrameworksFromProjectJson(manifestFile);
+    manifestFile = JSON.parse(manifestFileContents.trim());
   } catch (err) {
-    throw new Error(`Extracting target framework failed with error ${err.message}`);
+    throw new InvalidUserInputError(`Failed to parse manifest as valid json: ${err}`);
   }
+  return getTargetFrameworksFromProjectJson(manifestFile);
```
Comment (gitphill) [inline lib/index.ts]: this return was ok inside try, why has it moved?

Comment (lili2311) [reply lib/index.ts]: to isolate the error, I can move it back

Comment (dkontorovskyy) [reply lib/index.ts]: It will be more consistent with other methods. Or move all of them out or move all of them in, please

**Pull request closed without merging**